### PR TITLE
Cloud Monitoring: Remove dependencies

### LINKED ISSUE #79281: Extract GCM from grafana/grafana pt. 1

### PR DESCRIPTION
**Pull request by alyssajoyner**
This is part 1 of externalizing GCM from core grafana. It removes dependencies and fixes corresponding tests.

Closes #79281

### CODE REVIEW

File: public/app/plugins/datasource/cloud-monitoring/components/VisualMetricQueryEditor.test.tsx [hunk 1]
```diff
@@ -193,22 +271,24 @@ describe('VisualMetricQueryEditor', () => {
...
     const datasource = createMockDatasource({
       getMetricTypes: jest.fn().mockResolvedValue([createMockMetricDescriptor()]),
-      getLabels: jest
-        .fn()
-        .mockResolvedValue(
-          timeSrv.time.from === 'now-6h' ? { 'metric.test_groupby': '' } : { 'metric.test_groupby_1': '' }
-        ),
-      templateSrv: new TemplateSrv(),
-      timeSrv,
+      getLabels: jest.fn().mockResolvedValue({ 'metric.test_groupby_1': '' }),
```
Comment (aangelisc) [inline public/app/plugins/datasource/cloud-monitoring/components/VisualMetricQueryEditor.test.tsx:280]: I think the conditions of this test might be wrong now. This should be testing that the labels will change from `metric.test_groupby` to `metric.test_groupby_1`

File: public/app/plugins/datasource/cloud-monitoring/components/VisualMetricQueryEditor.test.tsx [hunk 2]
```diff
@@ -225,13 +305,19 @@ describe('VisualMetricQueryEditor', () => {
     const groupBy = await screen.findByLabelText('Group by');
     openMenu(groupBy);
     await waitFor(() => expect(document.body).toHaveTextContent('metric.test_groupby'));
-    timeSrv.setTime({ from: 'now-12h', to: 'now' });
+    // timeRange.from ({ from: 'now-12h', to: 'now' });
```
Comment (aangelisc) [inline public/app/plugins/datasource/cloud-monitoring/components/VisualMetricQueryEditor.test.tsx:308]: Commented code

File: public/app/plugins/datasource/cloud-monitoring/components/VisualMetricQueryEditor.tsx [hunk 3]
```diff
@@ -60,7 +62,7 @@ export function Editor({
     }
   };
 
-  useTime(datasource.timeSrv.timeRange());
+  useTime(getDefaultTimeRange());
```
Comment (aangelisc) [inline public/app/plugins/datasource/cloud-monitoring/components/VisualMetricQueryEditor.tsx:65]: Should this be `range`?

**Review by bossinc — APPROVED** [review]
:tada:

**Pull request merged**